x-pack/plugin/data-frame/src/test/java/org/elasticsearch/xpack/dataframe/DataFrameFeatureSetTests.java

```diff
@@ -142,16 +142,19 @@ public void testUsage() throws InterruptedException, ExecutionException, IOExcep
                 transformConfigWithoutTasks.forEach(ignored -> stateCounts.merge(IndexerState.STOPPED.value(), 1, Integer::sum));
                 stateCounts.forEach((k, v) -> assertEquals(v, XContentMapValues.extractValue("transforms." + k, usageAsMap)));
 
-                DataFrameIndexerTransformStats combinedStats = transformsStateAndStats.stream().map(x -> x.getTransformStats())
-                        .reduce((l, r) -> l.merge(r)).get();
-
-                assertEquals(toIntExact(combinedStats.getIndexFailures()),
-                        XContentMapValues.extractValue("stats.index_failures", usageAsMap));
-                assertEquals(toIntExact(combinedStats.getIndexTotal()), XContentMapValues.extractValue("stats.index_total", usageAsMap));
-                assertEquals(toIntExact(combinedStats.getSearchTime()),
-                        XContentMapValues.extractValue("stats.search_time_in_ms", usageAsMap));
-                assertEquals(toIntExact(combinedStats.getNumDocuments()),
-                        XContentMapValues.extractValue("stats.documents_processed", usageAsMap));
+                Optional<DataFrameIndexerTransformStats> combinedStats = transformsStateAndStats.stream().map(x -> x.getTransformStats())
+                        .reduce((l, r) -> l.merge(r));
+
+                if (combinedStats.isPresent()) {
```

Comment:
I think the problem of this test is on line 131, the `if` is broken, it broke because `transformConfigWithoutTasks` is empty but `transformsStateAndStats` isn't. 

I admit this isn't a master piece in test writing (I think I originally wrote it), it would be clearer and cleaner if we split this test into the empty and non-empty case. We then do not need to workaround here, which I am not sure about: it disables all asserts and tests nothing if `isPresent()` returns false, which means we have no coverage for this case.

Reply:
@hendrikmuhs looking at the test we do have assertions for when `combinedStats.isPresent()`, they all occur earlier in the test for asserting the number of DFs that are given in certain states. I am not sure what `else` tests would exist in this case. 

I should have caught this in my refactor,

Reply:
There are assertions on `transformConfigWithoutTasks` lines 143-144 even if `transformsStateAndStats` is empty. 

I pushed a small change so it asserts the stats have value 0 if `transformsStateAndStats` is empty